src/dev_volume.erl

```diff
@@ -156,6 +156,219 @@ mount(_M1, _M2, Opts) ->
             {error, ErrorMsg}
     end.
 
+%% @doc Check if the base device exists and if it does, check if the 
+%% partition exists.
+%% @param Device The base device to check.
+%% @param Partition The partition to check.
+%% @param PartitionType The type of partition to check.
+%% @param VolumeName The name of the volume to check.
+%% @param MountPoint The mount point to check.
+%% @param StorePath The store path to check.
+%% @param Key The key to check.
+%% @param Opts The options to check.
+%% @returns `{ok, Binary}' on success with operation result message, or
+%% `{error, Binary}' on failure with error message.
+-spec handle_mounting_partition(
+    term(), term(), term(), term(), term(), term(), term(), map()
+) -> {ok, binary()} | {error, binary()}.
+handle_mounting_partition(
+    Device, Partition, PartitionType, VolumeName, MountPoint, StorePath, 
+    Key, Opts
+) ->
+    ?event(debug_volume, 
+        {handle_mounting_partition, entry}
+    ),
+
+    maybe
+        {check_does_device_exist, ok} ?= {check_does_device_exist, check_does_device_exist(Device)},
+        {maybe_create_and_format_partition, ok} ?= {maybe_create_and_format_partition, maybe_create_and_format_partition(Device, Partition, PartitionType, Key)},
+        {mount_volume, ok} ?= {mount_volume, mount_volume(Partition, Key, MountPoint, VolumeName)},
+        {update_node_store, {ok, NewStore}} ?= {update_node_store, update_node_store(StorePath, Opts)},
+        {update_application_config, {ok, Result}} ?= {update_application_config, update_application_config(StorePath, NewStore, Opts)},
```

Comment:
This is neat but can be improved on legibility a bit more matching `{_, ok} ?=` once the atom is only relevant for error handling.